crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -208,6 +208,7 @@ pub fn impl_param_set(_input: TokenStream) -> TokenStream {
                 type State = (#(#param::State,)*);
                 type Item<'w, 's> = ParamSet<'w, 's, (#(#param,)*)>;
 
+                #[allow(non_snake_case)]
```

Comment:
Can you add a comment here about why this is required?

Reply:
Done, added two comments so one can know why the allow and where are the warning